src/components/ProgressBar/index.js

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const ProgressBar = (props) => {
+  const {
+    value,
+    max,
+    theme,
+    size,
+    label,
+    showPercentage,
+    percentagePosition,
+    percentageText,
+    className,
+    id,
+    ...rest
+  } = props;
+
+  const percentage = Math.min(Math.max(0, (value / max) * 100), 100);
+
+  const getCssClasses = () => {
+    const cssClasses = ["tyk-progress-bar"].concat(
+      className ? className.split(" ") : []
+    );
+    const sizeClass = size
+      ? `tyk-progress-bar--${size}`
+      : "tyk-progress-bar--md";
+    const themeClass = theme
+      ? `tyk-progress-bar--${theme}`
+      : "tyk-progress-bar--primary";
+
+    cssClasses.push(sizeClass);
+    cssClasses.push(themeClass);
+
+    return cssClasses.join(" ");
+  };
```

Comment:
could also write:
```suggestion
  const getCssClasses = () => [
    'tyk-progress-bar',
    className,
    `tyk-progress-bar--${size ?? 'md'}`
    `tyk-progress-bar--${theme ?? primary}`
  ].filter(Boolean).join(' ');
```
and the default values for `size` and `theme` would probably be better to be set when destructuring the props